### DISCUSSION TITLE
focus and editing when pressing once in costume component cell

### DISCUSSION THREAD

**Discussion by OmerSetty** [nick-keller/react-datasheet-grid#162] [Q&A]
I created a costume component but i have a problem.
There is a difference in its behaviour from a regular text component.
I see in your examples that when u click on a cell / tab to it, when you type the focus moves to the specific cell and the letters register as well.
In hebrew on the other hand when you move on a cell and type it doesnt register and there is no focus. only on the second time  you click on the cell there is a focus and u can type in it.
Well i dont really have a problem with it even tho it would be great if it was only one click to change the value loke in english.
The problem is when i create a costume component.
When i move to a cell (clicking it or tab to it), when i type not only that it doesnt register inside, and the is no focus, but it countinue the typing in the last cell that i was focused on.
Another click on the cell will make a focus, but still when you type it doest register, and again its changing the value of the cell that i was last focus on. 
Only on the third click now i can change the value of a current cell.
What do i need to add to my costume column / costume component to make to make it like your already built text component so when i click a cell and start typing no value of previus cells will change but only the current one

**Comment by nick-keller** [comment]
Hey @OmerSetty , could you please share some screen recordings?
Basically it starts editing as you type if you press any character that matches this regexp: `/^[ -~]$/`
I can update the regexp to include more characters

**Reply by OmerSetty** [reply]
hi first of all thank you for answering :)
well, at your basic text component it does start editing when you press these regex and any letter in the english alphabet.
the problem is in my costume column and component, there i need to press 3 times at the cell so it will start editing, if i dont do these 3 presses it will continue editing the previous cell, i guess i need to add somthing to my costume column / component i just dont know what
